File: app_pojav_zh/src/main/java/com/movtery/pojavzh/ui/dialog/SelectRuntimeDialog.java
package com.movtery.pojavzh.ui.dialog;

import android.content.Context;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import net.kdt.pojavlaunch.R;
import net.kdt.pojavlaunch.multirt.RTRecyclerViewAdapter;

public class SelectRuntimeDialog extends FullScreenDialog {
    private RecyclerView recyclerView;

    public SelectRuntimeDialog(@NonNull Context context) {
        super(context);

        this.setCancelable(false);
        this.setContentView(R.layout.dialog_select_item);
        init(context);
    }

    private void init(Context context) {
        recyclerView = findViewById(R.id.zh_select_view);
        TextView mTitleText = findViewById(R.id.zh_select_item_title);
        mTitleText.setText(R.string.multirt_config_title);
        ImageButton mCloseButton = findViewById(R.id.zh_select_item_close_button);

        mCloseButton.setOnClickListener(v -> this.dismiss());
        recyclerView.setLayoutManager(new LinearLayoutManager(context));
    }

    public void setListener(RuntimeSelectedListener listener) {
        RTRecyclerViewAdapter adapter = new RTRecyclerViewAdapter(listener);
        recyclerView.setAdapter(adapter);
    }

    public interface RuntimeSelectedListener {
        void onSelected(String jreName);
    }
}

File: app_pojav_zh/src/main/java/com/movtery/pojavzh/ui/dialog/SelectRuntimeDialog.kt
package com.movtery.pojavzh.ui.dialog

import android.content.Context
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import net.kdt.pojavlaunch.R
import net.kdt.pojavlaunch.multirt.MultiRTUtils
import net.kdt.pojavlaunch.multirt.RTRecyclerViewAdapter
import net.kdt.pojavlaunch.multirt.Runtime

class SelectRuntimeDialog(context: Context) : AbstractSelectDialog(context) {
    private var recyclerView: RecyclerView? = null

    init {
        this.setCancelable(false)
    }

    override fun initDialog(recyclerView: RecyclerView) {
        this.recyclerView = recyclerView
        setTitleText(R.string.install_select_jre_environment)
        setMessageText(R.string.install_recommend_use_jre8)
        recyclerView.layoutManager = LinearLayoutManager(context)
    }

    fun setListener(listener: RuntimeSelectedListener?) {
        val runtimes: MutableList<Runtime> = ArrayList(MultiRTUtils.getRuntimes())
        if (runtimes.isNotEmpty()) runtimes.add(Runtime("auto"))
        val adapter = RTRecyclerViewAdapter(runtimes, listener)
        recyclerView?.adapter = adapter
    }

    fun interface RuntimeSelectedListener {
        fun onSelected(jreName: String?)
    }
}
